### PR TITLE
tgui: Biogenerator

### PR DESCRIPTION
**Pull request by Arkatos1**
## About The Pull Request

This PR converts Biogenerator from HTML based interface to tgui, with some improvements and cleanup of the related code. You can now set custom multiplier for all items in the biogenerator, you are not limited to only some items and only in 5x/10x batches. When setting a multiplier of any item, the cost of an item is automatically adjusted to a new value, so you always see relevant information.

<details>
  <summary>Old UI</summary>

BiogenOLD

</details>

**NEW (Updated!):**

BiogenNEW2

## Why It's Good For The Game

Better UI.

## Changelog
:cl: Arkatos
add: Biogenerator now uses tgui.
/:cl:

### CODE REVIEW

**Comment by actioninja** [comment]
Ooo nice

**Review by stylemistake — COMMENTED** [review]
`useLocalState` inside of a loop caught my attention, since in normal React code that would be a bug (hooks have to maintain same call order), but  it's fine with our API since it's not order dependent. I have mixed feelings about this.

UI looks great, I love it.

File: tgui/packages/tgui/interfaces/Biogenerator.js [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    processing,
+  } = data;
+  return (
+    <Window resizable>
+      <Window.Content scrollable>
+        {!beaker && (
+          <NoticeBox>No Container</NoticeBox>
+        )}
+        {!!processing && (
+          <Dimmer fontSize="32px">
```
Comment (stylemistake) [inline tgui/packages/tgui/interfaces/Biogenerator.js:24]: Pull Dimmer outside of window content to make it not scroll with it.

Comment (Arkatos1) [reply tgui/packages/tgui/interfaces/Biogenerator.js]: Done.

**Pull request merged**